### PR TITLE
docs: new experimental Kafka scaler

### PR DESCRIPTION
**Pull request by sansmoraxz**
Docs for experimental kafka scaler which is based on `kafka-go`

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO)

Fixes https://github.com/kedacore/keda/issues/4692 https://github.com/kedacore/keda/issues/3431

Relates to https://github.com/kedacore/keda/pull/4801

### CODE REVIEW

**Comment by sansmoraxz** [comment]
Weird. When I try to run locally, the file is being generated.

File: content/docs/2.12/scalers/kafka-x.md [hunk 1]
```diff
@@ -0,0 +1,338 @@
++++
+title = "Apache Kafka Experimental Scaler"
+availability = "v2.12+"
+maintainer = "Microsoft"
```
Comment (JorTurFer) [inline content/docs/2.12/scalers/kafka-x.md:4]:
```suggestion
maintainer = "Community"
```

**Review by JorTurFer — COMMENTED** [review]
Nice!
AFAIR, there are some gaps comparing this new scaler with the already existing scaler with some features not implemented yet. Could you document them as part of the scaler page? Thanks to it users can decide which flavor matches better with their requirements

**Comment by sansmoraxz** [comment]
Sure will add those to docs.

Also this file should probably be renamed. How should we approach this?
The sarama implementation file is named `apache-kafka.md`

**Review by zroubalik — APPROVED** [review]

**Pull request merged**